coderd/workspaceapps/db.go

```diff
@@ -422,6 +425,9 @@ func (p *DBTokenProvider) connLogInitRequest(ctx context.Context, w http.Respons
 	// this ensures that the status and response body are available.
 	var committed bool
 	return aReq, func() {
+		// We want to log/audit the connection attempt even if the request context has expired.
+		ctx, cancel := context.WithCancel(p.ctx)
```

Comment:
Suggestion: As a potential safeguard, we could consider a timeout context here, arbitrarily 15-30 seconds seems appropriate. But I can also see an argument for not doing it, up to you.